File: second_largest.py
# # This program is to find the second largest element in the array

def second_largest(arr):
    n=len(arr)
    res=-1
    largest=0
    for i in range(1,n):
        if arr[i]>arr[largest]:
            res=largest
            largest=i
        
        elif arr[i]!=arr[largest]:
            if res==-1 or arr[i]>arr[res]:
                res=i
    return res

print(second_largest([5,20,12,20,8]))

File: second_largest.cpp
#include <iostream>
using namespace std;

int second_largest(int arr[],int n)
{
    int res = -1;
    int largest = 0;
    for (int i = 0;i < n ;i++){
        if (arr[i] > arr[largest]){
            res = largest;
            largest = i;
        }
        else if (arr[i] != arr[largest]){
            if (res == -1 || arr[i] > arr[res])
                res = i;
        }
    }
    return res;
}

int main()
{
    int arr[]= {5,20,7,20,10};
    cout << second_largest(arr,5);
    return 0;
}